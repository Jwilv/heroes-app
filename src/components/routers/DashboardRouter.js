import React from 'react'
import { Navbar } from '../ui/NavBar'
import {
    Routes,
    Route,
} from 'react-router-dom'
import { MarverlScreen } from '../marvel/MarverlScreen'
import { HeroesScreen } from '../heroes/HeroesScreen'
import { DcScreen } from '../dc/DcScreen'


export const DashboardRouter = () => {
    return (
        <>
            <Navbar />
            <div className='container mt-2'>
                <Routes>
                    <Route path='/' element={<HeroesScreen />} />
                    <Route path='/marvel' element={<MarverlScreen />} />
                    <Route path='/heroe/:heroeId' element={<HeroesScreen />} />
                    <Route path='/dc' element={<DcScreen />} />
                    <Route path='*' element={<h1>404 no page</h1>} />
                </Routes>
            </div>
        </>
    )
}
