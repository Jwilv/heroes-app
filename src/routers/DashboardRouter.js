import React from 'react'
import { Navbar } from '../components/ui/NavBar'
import {
    Routes,
    Route,
} from 'react-router-dom'
import { MarverlScreen } from '../components/marvel/MarverlScreen'
import { HeroesScreen } from '../components/heroes/HeroesScreen'
import { DcScreen } from '../components/dc/DcScreen'
import { SearchScreen } from '../components/search/SearchScreen'


export const DashboardRouter = () => {
    return (
        <>
            <Navbar />
            <div className='container mt-2'>
                <Routes>
                    <Route path='/' element={<MarverlScreen />} />
                    <Route path='/marvel' element={<MarverlScreen />} />
                    <Route path='/hero/:heroId' element={<HeroesScreen />} />
                    <Route path='/dc' element={<DcScreen />} />
                    <Route path='/search' element={<SearchScreen />} />
                    <Route path='*' element={<h1>404 no page</h1>} />
                </Routes>
            </div>
        </>
    )
}
