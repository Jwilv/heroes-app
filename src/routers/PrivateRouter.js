import React from 'react'
import PropTypes from 'prop-types'
import { Route, useNavigation } from 'react-router-dom'

export const PrivateRouter = ({
    isAuthenticated,
    component: Component,
    ...rest
}) => {
    const navigation = useNavigation();
    return (
        <Route
            {...rest}
            component={(props) => (
                (isAuthenticated)
                    ? (<Component {...props} />)
                    : <navigation to='/login'/>
        )}
        />
    )
}

PrivateRouter.propTypes ={
    isAuthenticated: PropTypes.bool.isRequired,
    component: PropTypes.func.isRequired,
}