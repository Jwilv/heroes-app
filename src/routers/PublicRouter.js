import React, { useContext } from 'react'
import PropTypes from 'prop-types'
import { Navigate } from 'react-router-dom'
import { AuthContext } from '../auth/AuthContext'

export const PublicRouter = ({ children }) => {

    const { user } = useContext(AuthContext)
    return (!user.logged)
        ? children
        : <Navigate to={'/marvel'} />
}

PrivateRouter.propTypes = {
    children: PropTypes.element.isRequired,
}