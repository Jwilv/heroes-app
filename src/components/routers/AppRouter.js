import React from 'react'
import {
    Routes,
    Route,
} from 'react-router-dom'
import { LoginScreen } from '../login/LoginScreen'
import { DashboardRouter } from './DashboardRouter'



export const AppRouter = () => {
    return (
        <>
            <Routes>
                <Route exact path='/login' element={<LoginScreen />} />
                 <Route  path='*' element={<DashboardRouter />} /> 
                {/* <Route path='*' element={<h1>404 alla la ponen en 4</h1>} /> */}


            </Routes>
        </>
    )
}