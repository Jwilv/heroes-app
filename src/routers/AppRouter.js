import React, { useContext } from 'react'
import {
    Routes,
    Route,
} from 'react-router-dom'
import { AuthContext } from '../auth/AuthContext'
import { LoginScreen } from '../components/login/LoginScreen'
import { DashboardRouter } from './DashboardRouter'
import { PrivateRouter } from './PrivateRouter'



export const AppRouter = () => {

    const { user } = useContext(AuthContext)

    return (
        <>
            <Routes>
                <Route exact path='/login' element={<LoginScreen />} />
                <Route path='*' element={
                    <PrivateRouter user={user} >
                        <DashboardRouter />
                    </PrivateRouter>
                }
                />
            </Routes>
        </>
    )
}