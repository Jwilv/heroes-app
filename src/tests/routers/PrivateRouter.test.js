import React from 'react'
import { render, shallow } from 'enzyme'
import { AuthContext } from "../../auth/AuthContext"
import { PrivateRouter } from "../../routers/PrivateRouter"
import { demoAuthTodo } from "../fixtures/demoAuthTodo"


describe('pruebas en <PrivateRouter />', () => {

    const { user } = demoAuthTodo[0]
    const Title = () => {
        return <h1>hola mundo</h1>
    }

    const authenticated = shallow(
        <AuthContext.Provider value={{ user }}>
            <PrivateRouter children={<Title />} />
        </AuthContext.Provider>);


    test('debe de mostrar el componente si esta autenticado', () => {
        expect(authenticated.find('Title').exists() ).toBe(true)
    })

    test('debe de mostrar el componente si esta autenticado', () => {
        expect(authenticated).toMatchSnapshot()
    })
})