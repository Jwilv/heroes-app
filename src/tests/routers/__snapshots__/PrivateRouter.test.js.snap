// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <PrivateRouter /> debe de mostrar el componente si esta autenticado 1`] = `
<PrivateRouter>
  <Title />
</PrivateRouter>
`;
